### PR TITLE
zsh-z: init at unstable-2021-02-15

### PR DESCRIPTION
**Pull request by evalexpr**
###### Motivation for this change

ZSH-z is a ZSH script port of [rupa/z](https://github.com/rupa/z).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/zsh/zsh-z/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "zsh-z";
+  version = "2021-02-15";
```
Comment (evalexpr) [inline pkgs/shells/zsh/zsh-z/default.nix:5]: Should I mark this as `unstable-2021-02-15`?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120733` at 3efd20e2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>zsh-z</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/shells/zsh/zsh-z/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -11378,4 +11378,14 @@
...
+  evalexpr = {
+    name = "Jonathan Wilkins";
+    email = "nixos@wilkins.tech";
+    github = "evalexpr";
+    githubId = 23485511;
+    keys = [{
+      longkeyid = "rsa4096/0x2D1D402E17763DD6";
+      fingerprint = "8129 5B85 9C5A F703 C2F4  1E29 2D1D 402E 1776 3DD6";
+    }];
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:11390]: This should be in another commit `maintainers: add Jonatah Wilkins`

File: pkgs/shells/zsh/zsh-z/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  installPhase = ''
+    mkdir -p $out/share/zsh-z
+    cp _zshz zsh-z.plugin.zsh $out/share/zsh-z
+  '';
+
+  meta = with lib; {
+    description = "Jump quickly to directories that you have visited frequently in the past, or recently";
+    homepage = "https://github.com/agkozak/zsh-z";
+    license = licenses.mit;
+
```
Comment (legendofmiracles) [inline pkgs/shells/zsh/zsh-z/default.nix:25]:
```suggestion
```

File: pkgs/shells/zsh/zsh-z/default.nix (hunk 1 above)
Comment (legendofmiracles) [inline pkgs/shells/zsh/zsh-z/default.nix:5]: Should maybe be renamed to `unstable-2021-02-15`, as everything without a release, or especially before one can be considered that

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

File: pkgs/shells/zsh/zsh-z/default.nix (hunk 1 above)
Comment (evalexpr) [reply pkgs/shells/zsh/zsh-z/default.nix:5]: Yeah I was thinking this too, will rename it

**Comment by evalexpr** [comment]
> 1 package built successfully:
> 1 suggestion:
> 
>     * warning: [missing-phase-hooks](https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md?rgh-link-date=2021-04-26T12%3A41%3A26Z)
>       `installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.
>       Near pkgs/shells/zsh/zsh-z/default.nix:16:3:
>       ```
>          |
>       16 |   installPhase = ''
>          |   ^
>       ```

I'm unsure if this is relevant or not, I guess so? though the other zsh packages in this directory don't contain these lines.

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -11378,4 +11378,14 @@
     github = "zupo";
     githubId = 311580;
   };
+  evalexpr = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11381]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

Comment (evalexpr) [reply maintainers/maintainer-list.nix]: This is already in a separate commit, do you mean a separate PR?

None of the maintainers above my one are in alphabetical order FWIW but I have put my one where it belongs.

File: pkgs/shells/zsh/zsh-z/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-z/default.nix:3]:
```suggestion
stdenvNoCC.mkDerivation rec {
```

File: pkgs/shells/zsh/zsh-z/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-z/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, fetchFromGitHub }:
```

**Review by SuperSandro2000 — COMMENTED** [review]
We don't compile anything so we don't need a stdenv with a compiler.
